### PR TITLE
Fix heatmap `norm=None` issue

### LINKED ISSUE #3275: `heatmap(..., norm=None, ...)` has different behaviour than without specifying the `norm` argument
Hi,

I'm noticing a strange behaviour when passing `norm=None` to a heatmap and I believe it's due to these lines: https://github.com/mwaskom/seaborn/blob/3733590d86a7f2c2a95cd9940a34aa7df5f5a3d2/seaborn/matrix.py#L299-L303

Specifically, if I use `sns.heatmap(..., vmin=0.0, vmax=1.0, ...)` I get something like this:

without-norm

but when I use `sns.heatmap(..., vmin=0.0, vmax=1.0, norm=None, ...)`, `vmin` and `vmax` are lost:

with-norm

I'm happy to send a PR if this issue isn't addressed anywhere.

Ștefan

### PR DESCRIPTION
**Pull request by stefanistrate**
Resolves #3275.

### CODE REVIEW

File: seaborn/matrix.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def plot(self, ax, cax, kws):
 
         # setting vmin/vmax in addition to norm is deprecated
         # so avoid setting if norm is set
-        if "norm" not in kws:
+        if "norm" not in kws or not kws["norm"]:
```
Comment (mwaskom) [inline seaborn/matrix.py:301]: This depends on `Norm` objects having a positive "truthiness", which may work in practice but feels undefined from first principles. I suggest this more explicit check:

```suggestion
        if kws.get("norm") is not None:
```

**Comment by mwaskom** [comment]
Thanks; seems reasonable aside from one nit, though a test would be nice too...

**Comment by stefanistrate** [comment]
> Thanks; seems reasonable aside from one nit, though a test would be nice too...

Thanks for suggestion. I'll add a test soon.

File: seaborn/matrix.py (hunk 1 above)
Comment (stefanistrate) [reply seaborn/matrix.py]: FYI, just double-checked your original comment now and I believe the equivalent to my statement would be `if kws.get("norm") is None`.

**Comment by stefanistrate** [comment]
I've just added a test that fails before my change and passes after. Please take another look.

**Comment by mwaskom** [comment]
Test looks perfect, thanks!

**Pull request merged**